### PR TITLE
Revert "feature #30501 [FrameworkBundle][Routing] added Configurators to handle template and redirect controllers (HeahDude)"

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

This reverts commit 477ee19778db2a30ac04d1dc1b6b32492ccf9f52, reversing changes made to 9bfa25869adab00162c246f4b76d65ca3b78e41a.

The reverted PR is #30501. Actually, it's a partial revert: changes made to the `Routing` component are not reverted.

The reason for this revert is that a discussion is still on-going in #35653 (which provides the same configuration extensions to yaml+xml), and we won't be able to resolve them in time for 5.1.

Better postpone for 5.2.

@HeahDude I invite you to open a PR after this one, reverting this very PR, which we'll consider again but for 5.2 if you don't mind.

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -83,7 +83,6 @@
         "symfony/messenger": "<4.4",
         "symfony/mime": "<4.4",
         "symfony/property-info": "<4.4",
-        "symfony/routing": "<5.1",
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/composer.json:86]: should not be re-reverted in the follow-up PR for 5.2

File: src/Symfony/Component/Routing/Loader/XmlFileLoader.php [hunk 2]
```diff
@@ -104,33 +104,31 @@ public function supports($resource, string $type = null)
     /**
      * Parses a route and adds it to the RouteCollection.
      *
-     * @param \DOMElement $node     Element to parse that represents a Route
-     * @param string      $filepath Full path of the XML file being processed
+     * @param \DOMElement $node Element to parse that represents a Route
+     * @param string      $path Full path of the XML file being processed
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Routing/Loader/XmlFileLoader.php:108]: should not be re-reverted in the follow-up PR for 5.2 (same for all this file: the changes put some lines back in their previous state and are fine as is, to reduce the diff between branches)

**Review by nicolas-grekas — COMMENTED** [review]
Notes for you @HeahDude, when preparing the PR for 5.2

**Pull request merged**